.github/workflows/test.yml

```diff
@@ -38,28 +38,68 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: ${{ env[matrix.python-version] }}
-      - name: Install tox
-        run: pip install -U tox-factor
-      - name: Cache tox environment
-        # Preserves .tox directory between runs for faster installs
+
+      - name: Cache python/tox
         uses: actions/cache@v2
         with:
           path: |
             .tox
             ~/.cache/pip
-          key: v4-build-tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'gen-requirements.txt', 'dev-requirements.txt') }}
+            c:\users\runneradmin\appdata\local\pip\cache
+          key: v6-build-tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'gen-requirements.txt', 'dev-requirements.txt') }}
+
+      # <-- pypy+windows fixes for instrumentation deps 
+      # cryptography does not publish pre-built wheels for pypy+windows so pip will try to build the wheel
+      # itself.
+      - name: Detect Win+PyPy+Instrumentation
+        id: windows-pypy-check
+        if: ${{ matrix.os == 'windows-2019' && matrix.python-version == 'pypy3' && matrix.package == 'instrumentation' }}
+        run: echo "::set-output name=hit::true"
+
+      - name: Win+PyPy - Setup Environment
+        if: steps.windows-pypy-check.outputs.hit
+        run: |
+          echo "INCLUDE=C:\Program Files (x86)\OpenSSL-win32\include" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+          echo "LIB=C:\Program Files (x86)\OpenSSL-win32\lib" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
```

Comment:
Can this whole step be remove in favor of setting the `env:` for the "Win+PyPy - Pre-build cryptography" step? I imagine it's not needed for `choco install`

Reply:
It is needed for cryptography install *and* tox run steps both.